internal/controller/controlplaneauth/controller.go

```diff
@@ -163,7 +163,7 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 
 	t, err := c.tokens.Create(ctx, &tokens.TokenCreateParameters{
 		Attributes: tokens.TokenAttributes{
-			Name: cr.Spec.ForProvider.ControlPlaneName,
+			Name: fmt.Sprintf("%.64s", cr.Spec.ForProvider.ControlPlaneName+"-"+string(cr.UID)),
```

Comment:
Sec, need to double check the column wide for these tokens 👀.

Reply:
Looks like it's 255:
```
`name` VARCHAR(255) NOT NULL,
```
To be honest, I don't know how good of an error we might return from this. Given that you're simply attempting to add some entropy to the name, it may be worth applying a shortening step if the supplied name is close to 255 characters.